fetchai/ledger/api/__init__.py

```diff
@@ -58,6 +60,31 @@ def _get_or_set_default_time(d, key, default):
     return default
 
 
+def _pre_process_version(reported_version):
+    server_version = reported_version.lstrip('v')
+
+    # remove trailing git version if present
+    match = re.match(r'^(\d+\.\d+\.\d+)-(alpha|beta|rc)(\d+)-\d+-(g[a-f0-9]+)$', str(server_version))
+    if match is not None:
+        server_version = '{}-{}.{}+{}'.format(match.group(1), match.group(2), match.group(3), match.group(4))
+
+    match = re.match(r'^(\d+\.\d+\.\d+)-(alpha|beta|rc)(\d+)$', str(server_version))
```

Comment:
```suggestion
    # insert . between prerelease label and number to enable numerical (rather than lexical) comparison of number
    match = re.match(r'^(\d+\.\d+\.\d+)-(alpha|beta|rc)(\d+)$', str(server_version))
```